sdk/appconfiguration/app-configuration/src/featureFlag.ts

```diff
@@ -24,9 +15,13 @@ export const featureFlagPrefix = ".appconfig.featureflag/";
 export const featureFlagContentType = "application/vnd.microsoft.appconfig.ff+json;charset=utf-8";
 
 /**
- * Necessary fields for updating or creating a new feature flag.
+ * Value of a feature flag
  */
-export interface FeatureFlagParam extends ConfigurationSettingParam {
+export interface FeatureFlagValue {
+  /**
+   * Id for the feature flag.
+   */
+  id?: string;
```

Comment:
is it actually optional?

Reply:
Yes, it is optional.
If the user doesn't provide an id, we(SDK) add an id based on the key just like the portal does.